__tests__/index.test.js

```diff
@@ -192,3 +192,69 @@ describe('GET /api/articles/:article_id', () => {
     })
 })
 
+describe('/api/articles/:article_id/comments', () => {
+    test('POSTS a new comment to an article with article id - status 201', () => {
+
+        const newComment = {
+            username: 'icellusedkars',
+            body: 'Uneblievable stuff Geoff!'
+        }
+
+        const returnedComment = {
+            author: 'icellusedkars',
+            body: 'Uneblievable stuff Geoff!',
+            article_id: 9,
+            votes: 0
+        }
+
+        return request(app)
+            .post('/api/articles/9/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                console.log(body.newComment);
+                expect(body.newComment).toMatchObject(returnedComment)
+            })
+    });
+
+    test('responds with error message if article id is valid but does not exist', () => {
+
+         return request(app)
+            .get('/api/articles/999999/comments')
+            .expect(404)
+            .then(({ body }) => {
+                const { msg } = body;
+                expect(msg).toBe("no article found with matching id");
+            })
+    });
+
+    test('sends an appropriate status and error message when given an invalid id', () => {
+        return request(app)
+        .get('/api/articles/not-an-article/comments')
+        .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('Bad request');
+            });
+    });
+
+    test('returns error message when username does not exist - status 404', () => {
+
+        const newComment = {
+            username: 'spot_the_dog',
+            body: 'Uneblievable stuff Geoff!'
+        }
+
+        return request(app)
+            .post('/api/articles/9/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({ body }) => {
+                const { msg } = body;
+                expect(msg).toBe("username not found");
+            })
+    });
+
+ })
```

Comment:
There is a test missing for when for example the body property is missing i.e. `newComment = {username: "icellusedkars"}`